orc8r/cloud/go/mproto/mproto.go

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package mproto
+
+import (
+	"magma/orc8r/cloud/go/mproto/protos"
+
+	"github.com/golang/protobuf/proto"
+)
+
+// EncodeProtosDeterministic deterministically encodes a slice of protobuf messages,
+// indexed by a unique identifier string, utilizing encodeProtoDeterministic
+func EncodeProtosDeterministic(protoSlice map[string]proto.Message) ([]byte, error) {
```

Comment:
Update verbiage throughout: this isn't actually a slice of protos, it's a map. But we don't care so much about the particular form, just that it's an unordered set of uniquely-identified protos. So this would be sufficient
```suggestion
func MarshalManyDeterministic(protosByID map[string]proto.Message) ([]byte, error) {
```